cpp/src/json/json_path.cu

```diff
@@ -729,6 +737,15 @@ __device__ parse_result parse_json_path(json_state& j_state,
   };
   push_context(j_state, commands, false);
 
+  // First do JSON validation
```

Comment:
I think the general policy for cudf is to refrain from validating data, as outlined in the [libcudf does not introspect data](https://github.com/rapidsai/cudf/blob/branch-24.04/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#libcudf-does-not-introspect-data). 

Do you think it would be feasible to introduce a parameter within `get_json_object_options` named `validate_json_string`, which defaults to false? However, Spark could retain this option as true. We can also move the JSON parser parameters from `get_json_object_options` if they do not directly influence `get_json_object` especially when `validate_json_string` is false. This could involve creating separate options like `json_parser_options` or `json_validator_options`.